CMSgov/mint-app
# [EASI-2215] Read Only Discussion Subpage

# EASI-2215

## Changes and Description

- Extracted Modal functionality out from Discussion component and into its own component
    - and updated various components that needs that modal component functionality (i.e. `AskaQuestion` component)
- Render the Discussion component on the Read Only Subpages
- Conditional renders for Collaborators and Assessment users
- Update various tests

## Acceptance Criteria

- Reuse the task list discussion component
- For the read-only view, only the model team members and BSG/assessment can ask/answer questions. 
- Anyone else should have the uneditable discussion center as noted in the second Figma
- Testing

## Note:

- Discussion panel truncation was a bigger lift than the intention of this ticket. [A separate ticket](https://jiraent.cms.gov/browse/EASI-2408) has been created to tackle that
- `Ask a question` or `Answer` does not pop up a modal as per figma designs, but talking with @patrickseguraoddball and we both recognized that the way the `Discussion` component is coded, it will require a big lift to implement that functionality. 
    - Solution used: the body content changes and renders the `Ask a Question` and `Answer` prompts


## How to test this change

For Collaborators and Assessment Users
1. Pre-seed data and navigate to `Discussion` of the Read Only pages for any model plan
2. Ensure `Ask a Question` link is there
3. Submit several questions and answers 
4. See that the questions are rendering correctly

For **non**-collaborators and assessment users
1. Pre-seed data and navigate to `Discussion` of the Read Only pages for any model plan
2. See that no link is available 




## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
